### PR TITLE
Org mode improvements (extend markdownify to support all markup formats)

### PR DESCRIPTION
**Pull request by niklasfasching**
- document org front matter format
- introduce htmlify (#5589)

### CODE REVIEW

**Comment by bep** [comment]
We need to stop a minute and think a little about the name. I may have thought these "ify" function names was clever (I think we stole it from Jekyll), but I'm not so sure anymore. `markdownify` was obviously not fantastic.

I suggest we call the new one `Markup`.

That said, we also need to specify the block (paragraphs) vs inline behaviour of such func. There are miles of discussions about this  elsewhere, so I will not go into details.

**Comment by kaushalmodi** [comment]
I agree that the "ify" names are getting old.

> I suggest we call the new one Markup.

How about `toHTML` in `Markdown`, `Org`, .. namespaces?

Can we have something like:

```
{{ "**BatMan**" | Markdown.toHTML }}
```

```
{{ "*BatMan*" | Org.toHTML }}
```

I feel like above are readable as in plain English too.

**Comment by bep** [comment]
The current namespace is `transform.Markdownify` (I don't want 1 function namespaces) and I suggest we do `transform.Markup`. The name is pretty good, and another reason is that I plan to add  `.Page.Markup` method which defaults to the renderer or the page itself. It would be good if they shared the same name (and we're not adding a bunch of new methods to the alrady big Page interface).

**Comment by regisphilibert** [comment]
I'm sorry I don't have any suggestion here. But for me Markup in the global sense is not HTML only, it's an ensemble of factors determining the look of your page. CSS is part of the "Markup".

**Comment by kaushalmodi** [comment]
@regisphilibert As an aside.. that you mention:

> Originally these the little handwritten comments added next to the text to indicate the publisher how those element should be "styled" on once printed.

https://shadycharacters.co.uk/ is an awesome read if you are interested in such topics related to publishing and special characters 😃

**Comment by niklasfasching** [comment]
>  I suggest we do transform.Markup
> So.. how about a middle ground.. transform.ToHTML

I don't care either way but I'm unsure where to go from here.

**Comment by kaushalmodi** [comment]
Given that no one has a preference, but I do, @bep, does my proposal in https://github.com/gohugoio/hugo/pull/6043#issuecomment-502778769 sound good? 🙂

**Comment by bep** [comment]
>Given that no one has a preference, 
I have a preference, which is the more general term mentioned earlier.

* When I say `.Markup` (which is the page contextual variant of what we discuss here) I say: Render this using the page's rendering engine to the current output format.
* I understand that what we currently have renders to HTML only, I would be rather disappointed if the future did not bring ... more.

**Comment by bep** [comment]
@kaushalmodi @niklasfasching sorry, I have been slightly distracted lately. I think I will delay parts of this discussion to a later time. For this particular PR I will say that we cannot add another function with a different name but with almost identical functionality (you have no idea how much time just explaining the difference to people will add up to, even if we add fantastic docs).

So, also saying that in the future we need something to indicate block vs inline for this, I suggest that we:

1. Keep the name `markdownify` for now. We can create an alias for it once we know what that is.
2. Add an optional options `map` so I can do `{{ "=hello=" | markdownify (dict "format" "org") }}`

The above will map fairly nicely to the `.Markdownify` func, which then uses the page context to guess the above.

**Comment by bep** [comment]
Bump.

**Comment by niklasfasching** [comment]
Got distracted - I'll look into this on the weekend :)

**Comment by bep** [comment]
I will probably replace this PR now ...

I have thought about this, and I think `.Page.ToOutputFormat` (which will fall back to HTML for now) would be an OK name. It's a little bit mouthy, but I think it describes what it does well.

/cc @regisphilibert

**Comment by regisphilibert** [comment]
I'm a bit confused here as of the purpose of this `.ToOutputFormat` method. Its name evokes compatibility with multiple output. I understand it can take either markdown or org as input but I'm not sure I can figure what output, other than HTML, it could ever produce?

If I'm working on a JSON template file and I use `.Page.ToOutputFormat`, what does happen?

**Comment by bep** [comment]
@regisphilibert I changed my mind as I wrote it ... See my update. You are too fast.

**Comment by regisphilibert** [comment]
Yes `.RenderString` is great! And now I see we're talking about Rendering a string with shortcode, markdown etc... This will help greatly building pages with Front Matter Blocks! 🎉
